### PR TITLE
EDU-976 Prevent password brute forcing

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-976

### CODE REVIEW

File: src/server/admin-controller.js [hunk 1]
```diff
@@ -118,6 +123,12 @@ class AdminController {
       needsPermission(permissions.ADMIN),
       (req, res) => this.handlePostCdnUploadDirectoryCreationRequest(req, res)
     );
+
+    router.get(
+      '/api/v1/admin/request-info',
+      needsPermission(permissions.ADMIN),
+      (req, res) => this.handleGetRequestInfoRequest(req, res)
+    );
```
Comment (ahelmberger) [inline src/server/admin-controller.js:131]: This new endpoint is there to find out the number of proxies the application is running behind on production (see also the  explanation in the ticket: https://educandu.atlassian.net/browse/EDU-976)

File: src/services/request-limit-record-service.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import Logger from '../common/logger.js';
+import { createShortHash } from '../utils/string-utils.js';
+import RequestLimitRecordStore from '../stores/request-limit-record-store.js';
+
+const logger = new Logger(import.meta.url);
+
+const FAILED_LOGIN_ATTEMPTS_THRESHOLD = 5;
+const FAILED_LOGIN_ATTEMPTS_KEY = 'failed-login-attempts';
+const FAILED_LOGIN_ATTEMPTS_TIME_WINDOW_IN_MS = 15 * 60 * 1000;
```
Comment (adrianaluchian) [inline src/services/request-limit-record-service.js:9]: I know these are used juts here, but maybe it would be handy to collect all such setup in the `constants` module. We already have similar constants there like `PENDING_USER_REGISTRATION_EXPIRATION_IN_HOURS`

File: src/services/request-limit-record-service.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    return [RequestLimitRecordStore];
+  }
+
+  constructor(requestLimitRecordStore) {
+    this.requestLimitRecordStore = requestLimitRecordStore;
+  }
+
+  getRoomById(roomId) {
+    return this.roomStore.getRoomById(roomId);
+  }
```
Comment (adrianaluchian) [inline src/services/request-limit-record-service.js:22]: This is probably a leftover.

File: src/services/request-limit-record-service.js (hunk 2 above)
Comment (ahelmberger) [reply src/services/request-limit-record-service.js]: Good point. Moved them to the `constants` module.

File: src/services/request-limit-record-service.js (hunk 3 above)
Comment (ahelmberger) [reply src/services/request-limit-record-service.js]: :scream: You should have never seen that.

File: src/utils/string-utils.spec.js [hunk 4]
```diff
@@ -49,4 +49,12 @@ describe('string-utils', () => {
     });
   });
 
+  describe('createShortHash', () => {
+    it('returns the same hash code for the same input', () => {
```
Comment (adrianaluchian) [inline src/utils/string-utils.spec.js:53]: I think this test can only be trusted if there's also a test that checks the negative scenario 😋

Comment (ahelmberger) [reply src/utils/string-utils.spec.js]: The function and the test are gone now anyways :-)

File: src/components/confirmation-dialogs.js [hunk 5]
```diff
@@ -283,6 +283,7 @@ export function reloginAfterSessionExpired(modal, t, onOk, onCancel) {
   const formRef = createRef();
 
   let dialog = null;
+  let isFrozen = false;
```
Comment (adrianaluchian) [inline src/components/confirmation-dialogs.js:286]: This variable should maybe have a clearer name since the module is not a "login" dedicated context. Maybe `isLoginFrozen` or `isLoggingInFrozen` to correlate with `isLogginIn`?

File: src/services/request-limit-record-service.js (hunk 3 above)
Comment (adrianaluchian) [reply src/services/request-limit-record-service.js]: 😵 🙈

File: src/utils/string-utils.spec.js (hunk 4 above)
Comment (adrianaluchian) [reply src/utils/string-utils.spec.js]: 👍

File: src/components/confirmation-dialogs.js (hunk 5 above)
Comment (ahelmberger) [reply src/components/confirmation-dialogs.js]: Agree. Changed it to `isLoginFrozen`.

**Review by adrianaluchian — APPROVED** [review]
